cli/cmd/report_rules.go

```diff
@@ -0,0 +1,399 @@
+
+	for _, key := range keys {
+		enabled := strings.Title(strconv.FormatBool(notifcationsMap[key]))
+		notifications = append(notifications, []string{key, enabled})
+	}
+
+	detailsTable.WriteString(renderCustomTable([]string{"NOTIFICATION TYPES", "ENABLED"}, notifications,
+		tableFunc(func(t *tablewriter.Table) {
+			t.SetBorder(false)
+			t.SetColumnSeparator(" ")
+			t.SetAutoWrapText(false)
+		}),
+	),
+	)
+	detailsTable.WriteString("\n")
+
+	if len(rule.EmailAlertChannels) > 0 {
+		channels := [][]string{{strings.Join(rule.EmailAlertChannels, "\n")}}
+		detailsTable.WriteString(renderCustomTable([]string{"EMAIL ALERT CHANNELS"}, channels,
+			tableFunc(func(t *tablewriter.Table) {
+				t.SetBorder(false)
+				t.SetColumnSeparator(" ")
+			}),
+		),
+		)
+		detailsTable.WriteString("\n")
+	}
+
+	if len(rule.Filter.ResourceGroups) > 0 {
+		resourceGroups := [][]string{{strings.Join(rule.Filter.ResourceGroups, "\n")}}
+		detailsTable.WriteString(renderCustomTable([]string{"RESOURCE GROUPS"}, resourceGroups,
+			tableFunc(func(t *tablewriter.Table) {
+				t.SetBorder(false)
+				t.SetColumnSeparator(" ")
+			}),
+		),
+		)
+	}
+
+	return detailsTable.String()
+}
+
+func promptCreateReportRule() (api.ReportRuleResponse, error) {
+	channelList, channelMap := getEmailAlertChannels()
+	notificationFields := structs.Names(api.ReportRuleNotificationTypes{})
+	notificationsMap := make(map[string]bool)
+
+	if len(channelList) < 1 {
+		return api.ReportRuleResponse{}, errors.New("no email alert channels found.")
+	}
+
+	questions := []*survey.Question{
+		{
+			Name:     "name",
+			Prompt:   &survey.Input{Message: "Name: "},
+			Validate: survey.Required,
+		},
+		{
+			Name:     "description",
+			Prompt:   &survey.Input{Message: "Description: "},
+			Validate: survey.Required,
+		},
+		{
+			Name: "channels",
+			Prompt: &survey.MultiSelect{
+				Message: "Select email alert channels:",
+				Options: channelList,
+			},
+			Validate: survey.Required,
+		},
+		{
+			Name: "severities",
+			Prompt: &survey.MultiSelect{
+				Message: "Select severities:",
+				Options: []string{"Critical", "High", "Medium", "Low", "Info"},
+			},
+		},
+		{
+			Name: "notifications",
+			Prompt: &survey.MultiSelect{
+				Message: "Select report notification types:",
+				Options: notificationFields,
+			},
+		},
+	}
+
+	answers := struct {
+		Name           string
+		Description    string   `survey:"description"`
+		Channels       []string `survey:"channels"`
+		Severities     []string `survey:"severities"`
+		ResourceGroups []string `survey:"resourceGroups"`
+		Notifications  []string `survey:"notifications"`
+	}{}
+
+	err := survey.Ask(questions, &answers,
+		survey.WithIcons(promptIconsFunc),
+	)
+	if err != nil {
+		return api.ReportRuleResponse{}, err
+	}
+
+	var channels []string
+	for _, channel := range answers.Channels {
+		channels = append(channels, channelMap[channel])
+	}
+
+	resourceGroups, resourceGroupMap := promptAddResourceGroupsToReportRule()
```

Comment:
same here, if functions "can" return `nil` values, you will have to check them. One way to fix this is by always returning an empty array or something other than `nil`

Reply:
A range of nil is 0. In this scenario there is no issue. But I agree with the sentiment of nil checking. But here we're not trying to access a nil value